### PR TITLE
Enhance HTML Builder example

### PR DESCRIPTION
**Pull request by VasylTretiakov**
Possible fix for #123

### CODE REVIEW

**Comment by AlexanderPrendota** [comment]
Hey, thanks for PR. @p7nov Could u have a look please?

**Review by AlexanderPrendota — APPROVED** [review]

File: examples/09_Kotlin_JS/06_HtmlBuilder.md [hunk 1]
```diff
@@ -47,25 +43,61 @@ fun main(args: Array<String>) {
...
 }
 
 abstract class TagWithText(name: String) : Tag(name) {
     operator fun String.unaryPlus() {
-        element.appendChild(document.createTextNode(this + " "))
+        children.add(TextElement(this))
     }
 }
-
 class HTML() : TagWithText("html") {
```
Comment (p7nov) [inline examples/09_Kotlin_JS/06_HtmlBuilder.md:101]: The version with your update does not parse \<ul> and \<li> tags. 
Please add this function into HTML:
`fun ul(init: UL.() -> Unit) = initTag(UL(), init)`
and see comments below.

```suggestion
class HTML() : TagWithText("html") {
    fun ul(init: UL.() -> Unit) = initTag(UL(), init)
```

File: examples/09_Kotlin_JS/06_HtmlBuilder.md [hunk 2]
```diff
@@ -99,11 +131,11 @@ class LI() : BodyTag("li")
 class P() : BodyTag("p")
 class H1() : BodyTag("h1")
```
Comment (p7nov) [inline examples/09_Kotlin_JS/06_HtmlBuilder.md:132]: Add these classes for proper parsing of \<ul> and \<li>:

```suggestion
class H1() : BodyTag("h1")
class LI() : BodyTag("li")

class UL : BodyTag("ul") {
    fun li(init: LI.() -> Unit) = initTag(LI(), init)
}

```

**Review by p7nov — CHANGES_REQUESTED** [review]
Hey!
Sorry for long waiting, didn't have a chance to check your changes.

Thanks a lot for helping us keep the examples up to date, great job!
However, I've spotted a minor problem: the DSL example from the doc page doesn't have \<ul> and \<li> tags. Please see my suggestions on how to fix this.

Regards, Pavel
Kotlin Technical Writer

File: examples/09_Kotlin_JS/06_HtmlBuilder.md (hunk 2 above)
Comment (VasylTretiakov) [reply examples/09_Kotlin_JS/06_HtmlBuilder.md:133]: This snippet is already present in my source

File: examples/09_Kotlin_JS/06_HtmlBuilder.md [hunk 3]
```diff
@@ -7,10 +7,6 @@ Below is an example of a type-safe Groovy-style builder. In this example, we wil
 ```run-kotlin-canvas
 package html
 
-import org.w3c.dom.*
-import kotlin.browser.document
-import kotlin.browser.window
-
 fun main(args: Array<String>) {
```
[the remark this answers is not in the document]
Comment (VasylTretiakov) [reply examples/09_Kotlin_JS/06_HtmlBuilder.md]: Makes sense

File: examples/09_Kotlin_JS/06_HtmlBuilder.md (hunk 1 above)
Comment (VasylTretiakov) [reply examples/09_Kotlin_JS/06_HtmlBuilder.md:101]: This does not look right as HTML consists of **head** and **body** while the **ul** tag goes inside the **body**. I already included the **ul** fun into the abstract class BodyTag.
Unfortunately, I have accidentally merged the change, and now do not see how to revert this commit.

Comment (p7nov) [reply examples/09_Kotlin_JS/06_HtmlBuilder.md:101]: Yes, you're completely right. My bad, I haven't checked the whole file in your branch, just the changes. Sorry.
Let's just remove the redundant declarations in a separate commit. I'll squash all this PR into one commit with the right corrections when merging.
Please remove:
- `fun ul()` from `HTML` class
- second declarations of `ul` and `li` classes.

Comment (VasylTretiakov) [reply examples/09_Kotlin_JS/06_HtmlBuilder.md:101]: Please note that before submitting the PR I ensured that the code compiles and produces correct results.
I reverted all new commits and re-applied the first one with a meaningful name.

**Comment by p7nov** [comment]
Thank you for updating this page. If you have more improvement ideas or any questions, share them with us here or in [YouTrack](https://youtrack.jetbrains.com/issues/KT)
Have a nice Kotlin!

**Pull request merged**